app/admin/pages.rb

```diff
@@ -261,7 +262,21 @@
               para "Year: #{component&.published_on_year}" if component&.published_on_year.present?
               para "Authors: #{component&.authors}" if component&.authors.present?
             end
-
+            # Event
+            if pc.component_type == 'PageEventComponent'
+              fields = {
+                title: 'Title',
+                url: 'URL',
+                presented_by: 'Presented by',
+                start_date: 'Start date',
+                end_date: 'End date',
+                location: 'Location',
+                text: 'Description'
+              }
+              fields.each do | key, value|
+                para "#{value}: #{component.send(key)}" if component.send(key).present?
+              end
+            end
```

Comment:
I was starting to find this section of the code pretty unreadable, so I opted for an info by component approach. Let me know if you think this is worth doing in the rest of this file.

Reply:
I think this approach makes sense.. Do you think it would be more rails-y if we took it a step further and abstracted the `fields` to be within the individual models? We might then be able to just interpolate the `component_type` and iterate over it's `FIELDS`:

`class PageEventComponent < ApplicationRecord
  FIELDS = {
    title: 'Title',
    url: 'URL',
    presented_by: 'Presented by',
    start_date: 'Start date',
    end_date: 'End date',
    location: 'Location',
    text: 'Description'
  }.freeze
end`

`component_type = pc.component_type.constantize
  if component_type.respond_to?(:FIELDS)
    component_type::FIELDS.each do |key, value|
      para "#{value}: #{pc.send(key)}" if pc.send(key).present?
    end
  end`

Reply:
I think this approach makes sense.. Do you think it would be more rails-y if we took it a step further and abstracted the fields to be within the individual models? We might then be able to just interpolate the component_type and iterate over it's FIELDS:

<img width="399" alt="Screenshot 2023-07-31 at 11 28 58 AM" src="https://github.com/department-of-veterans-affairs/diffusion-marketplace/assets/65036872/31c2222f-5b15-45a9-8db4-c91d02633eb1">

<img width="533" alt="Screenshot 2023-07-31 at 11 29 13 AM" src="https://github.com/department-of-veterans-affairs/diffusion-marketplace/assets/65036872/0835643f-d3bc-4b81-a2d0-6d2154cd3498">

Reply:
I like it! I'll use the pattern for Events here and add a separate PR for refactoring the other components